snakefile

```diff
@@ -99,6 +99,62 @@ rule extract_geoboundaries:
     shell:
         "extract_geoboundaries"
 
+## OSM Extract ##
+
+rule extract_osm_rivers:
+    input:
+        os.path.join(
+            config['dirs']['schemas'], config['osm']['rivers']['osm_tags'])
+    params:
+        url=config['osm']['url'],
+        country=config['constants']['ISO2']
+    output:
+        os.path.join(
+            config['dirs']['raw_data'], config['osm']['rivers']['raw'])
+    shell:
+        "extract_osm \"{params.url}\" \"{params.country}\" {input} {output}"
+
+rule extract_osm_lakes:
+    input:
+        os.path.join(
+            config['dirs']['schemas'], config['osm']['lakes']['osm_tags'])
+    params:
+        url=config['osm']['url'],
+        country=config['constants']['ISO2']
+    output:
+        os.path.join(
+            config['dirs']['raw_data'], config['osm']['lakes']['raw'])
+    shell:
+        "extract_osm \"{params.url}\" \"{params.country}\" {input} {output}"
+
+rule extract_osm_admin:
+    input:
+        os.path.join(
+            config['dirs']['schemas'], config['osm']['admin']['osm_tags'])
+    params:
+        url=config['osm']['url'],
+        country=config['constants']['ISO2']
+    output:
+        os.path.join(
+            config['dirs']['raw_data'], config['osm']['admin']['raw'])
+    shell:
+        "extract_osm \"{params.url}\" \"{params.country}\" {input} {output}"
+
+rule extract_osm_roads:
```

Comment:
Rule _extract_osm_roads_ similar to line 300 rule _extract_roads_osm_
Both use different methods.
Recommend removing old method _extract_roads_osm_ as new method more configurable (although new requests method does appear slower than old WGET extract method).

New transformation rule not included. Recommend adapting old transformation rule _transform_roads_osm_ (line 319) to work with new extract rule.

Reply:
Looks like this is because I'm requesting the entire dataset into memory before writing.  I'll look into using the requests methods in the utils folder.